redis/stub_test.go

```diff
@@ -12,11 +14,26 @@ import (
 	"unicode"
 )
 
-// RunT starts a new redis stub for a given test context.
+// RunT starts a new redis stub TCP server for a given test context.
 // It registers the test cleanup after your test is done.
 func RunT(t testing.TB) *StubServer {
 	s := NewStubServer()
-	if err := s.Start(); err != nil {
+	if err := s.Start(false, nil); err != nil {
+		t.Fatalf("could not start RedisStub; reason: %s", err)
+	}
+
+	t.Cleanup(s.Close)
+
+	return s
+}
+
+// RunTSecure starts a new redis stub TLS server for a given test context.
+// It registers the test cleanup after your test is done.
+// Optionally, a client certificate in PEM format can be passed to enable TLS
+// client authentication (mTLS).
+func RunTSecure(t testing.TB, clientCert []byte) *StubServer {
```

Comment:
I didn't want to change `RunT()` to avoid updating all existing tests that use it, and to have a simpler signature, so I just added this secure variant. Let me know if you'd rather have a single function for both cases.

BTW, why is it called `RunT`? Wouldn't `StartStubServer` make more sense, since that's all it does?